packages/canary-react/src/component-tests/IcDataTable/IcDataTable.cy.tsx

```diff
@@ -5627,3 +5630,219 @@ describe("Dark mode", () => {
+    mount(
+      <div
+        style={{
+          padding: "0.5rem",
+        }}
+      >
+        <IcTheme theme="light">
+          <NativeTable />
+        </IcTheme>
+      </div>
+    );
+
+    cy.get("table").find("tr").eq(1).realHover().wait(300);
+
+    // snapshot does not capture hover color for some reason,
+    // so additional check that hover state css is applied to the row
+    cy.get("table")
+      .find("tr")
+      .eq(1)
+      .should("have.css", "background-image")
+      .and(
+        "eq",
+        "linear-gradient(rgba(23, 89, 188, 0.1) 0px, rgba(23, 89, 188, 0.1) 0px)"
+      );
+
+    cy.compareSnapshot({
+      name: "/native-table/light-mode",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.014),
+      cypressScreenshotOptions: {
+        capture: "viewport",
+      },
+    });
+  });
+
+  it("should render a dense native table", () => {
+    mount(
+      <div
+        style={{
+          padding: "0.5rem",
+        }}
+      >
+        <IcTheme theme="light">
+          <NativeTable type="dense" />
+        </IcTheme>
+      </div>
+    );
+
+    cy.compareSnapshot({
+      name: "/native-table/dense-light-mode",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.013),
+      cypressScreenshotOptions: {
+        capture: "viewport",
+      },
+    });
+  });
+
+  it("should render a spacious native table", () => {
+    mount(
+      <div
+        style={{
+          padding: "0.5rem",
+        }}
+      >
+        <IcTheme theme="light">
+          <NativeTable type="spacious" />
+        </IcTheme>
+      </div>
+    );
+
+    cy.compareSnapshot({
+      name: "/native-table/spacious-light-mode",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.013),
+      cypressScreenshotOptions: {
+        capture: "viewport",
+      },
+    });
+  });
+
+  it("should render an embedded native table", () => {
+    mount(
+      <div
+        style={{
+          padding: "0.5rem",
+        }}
+      >
+        <IcTheme theme="light">
+          <NativeTable type="embedded" />
+        </IcTheme>
+      </div>
+    );
+
+    cy.compareSnapshot({
+      name: "/native-table/embedded-light-mode",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.014),
+      cypressScreenshotOptions: {
+        capture: "viewport",
+      },
+    });
+  });
+
+  it("should render a native table in dark mode", () => {
+    mount(
+      <div
+        style={{
+          backgroundColor: "var(--ic-color-page-background-dark)",
+          padding: "0.5rem",
+        }}
+      >
+        <IcTheme theme="dark">
+          <NativeTable />
+        </IcTheme>
+      </div>
+    );
+
+    cy.get("table").find("tr").eq(1).realHover().wait(300);
+
+    // snapshot does not capture hover color for some reason,
+    // so additional check that hover state css is applied to the row
+    cy.get("table")
+      .find("tr")
+      .eq(1)
+      .should("have.css", "background-image")
```

Comment:
logic duplicated across test cases so they could either be joined or the selectors global vars.